packages/doenetml-worker-rust/lib-doenetml-core/src/components/types.rs

```diff
@@ -120,6 +166,7 @@ pub struct PropSource {
     /// For example, given `<textInput name="i"/>`, a direct ref would be `$i.value` by itself,
     /// unlike `<text extend="$i.value"/>`.
     ///
+    /// XXX: Incorrect comment?
```

Comment:
Right, it's an outdated comment. It is no longer done in the DataQuery but accomplished when we build the structure graph itself.

Attempt at a revised comment:
If we are extending from a direct ref,
we need to add the referenced prop as a child in the structure graph
because an actual child component for the prop is not created.